curator-client/src/test/java/org/apache/curator/BasicTests.java

```diff
@@ -30,15 +37,13 @@
 import org.apache.zookeeper.ZooDefs;
 import org.apache.zookeeper.ZooKeeper;
 import org.mockito.Mockito;
-import org.testng.Assert;
-import org.testng.annotations.Test;
 import java.util.concurrent.Callable;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 public class BasicTests extends BaseClassForTests
 {
-    @Test
+    @RepeatedIfExceptionsTest(repeats = BaseClassForTests.REPEATS)
```

Comment:
There are multiple ways to repeat failed tests with jUnit5.
Most simple is Maven's surefire plugin, which works now with jUnit5.
But in Curator in the original solution not all of the tests have been repeated when failed. Some of them have been tried only once, some not.
This extension provides an annotation which you can use on those tests which you want to repeat when failed, but still have the option to use @Test on those you are not interested in. I thinks this greater flexibility is valuable.
What do you think?

Reply:
Personally (non binding) I would prefer to use the standard tools from the jUnit framework + surefire.
So add a simply rerunFailingTests on surefire configuration is a better option.

It would be better to not have flaky tests at all, but that would be a different story

@Randgalt what's your opinion ?

Reply:
I agree that not having to annotate every test is a good idea. If this functionality can be provided by configuration at a higher level that sounds like a good approach to me.

Reply:
@Randgalt I you also prefer the maven based solution, I can change it, it's not a big deal.
The only difference is that then we cannot define on test level which tests should re-run if fail. But I don't see it as a problem.